src/utils.ts

```diff
@@ -47,10 +47,21 @@ export const deriveSponsorWallet = (
   //
   // For self-funded feeds it's more suitable to derive the hash also from update parameters. This does not apply to
   // mananaged feeds which want to be funded by the same wallet independently of the update parameters.
-  const sponsorAddressHash =
-    walletDerivationScheme.type === 'self-funded'
-      ? deriveSponsorAddressHashForSelfFundedFeed(dapiNameOrDataFeedId, updateParameters)
-      : deriveSponsorAddressHashForManagedFeed(dapiNameOrDataFeedId);
+  let sponsorAddressHash: Hex;
+  switch (walletDerivationScheme.type) {
+    case 'self-funded': {
+      sponsorAddressHash = deriveSponsorAddressHashForSelfFundedFeed(dapiNameOrDataFeedId, updateParameters);
+      break;
+    }
+    case 'managed': {
+      sponsorAddressHash = deriveSponsorAddressHashForManagedFeed(dapiNameOrDataFeedId);
+      break;
+    }
+    case 'fixed': {
+      sponsorAddressHash = walletDerivationScheme.sponsorAddress!;
```

Comment:
This is very awkward to read. I think the concept of `sponsorAddressHash` doesn't make sense anymore. It seems better to define `sponsorAddress` and do something like this:
```
    case 'self-funded': {
      const sponsorAddressHash = deriveSponsorAddressHashForSelfFundedFeed(dapiNameOrDataFeedId, updateParameters);
      sponsorAddress = ethers.getAddress(sponsorAddressHash.slice(0, 42)) as Address;
      break;
    }
    ...
```

Reply:
Also, the documentation needs to be updated for this change.

Reply:
Done [here](https://github.com/api3dao/airseeker-v2/pull/296/commits/ad7ac3aa2c0a0254572bccffba7f9ad2903c9a68) but I won't resolve this conversation until I update the documentation